analysis_notebooks/Geography Analysis.ipynb

```diff
@@ -306,154 +311,253 @@
+    "                subregion_counts[subregion] += count\n",
+    "            else:\n",
+    "                subregion_counts[subregion] = count\n",
     "        except KeyError:\n",
-    "            pass\n",
-    "#             print(\"{} not found in CountryInfo\".format(country))\n",
-    "    try:\n",
-    "        subregion = iso3_to_subregion[iso3]\n",
-    "        if subregion in subregion_counts.keys():\n",
-    "            subregion_counts[subregion] += count\n",
-    "        else:\n",
-    "            subregion_counts[subregion] = count\n",
-    "    except KeyError:\n",
-    "        print(\"This country's subregion not found: {}\".format(country))\n",
+    "            print(\"This country's subregion not found: {}\".format(country))\n",
+    "    for key in iso3_to_scaledpop.keys():\n",
+    "        iso3_to_scaledpop[key] /= min(iso3_to_scaledpop.values())\n",
     "\n",
-    "for key in iso3_to_scaledpop.keys():\n",
-    "    iso3_to_scaledpop[key] /= min(iso3_to_scaledpop.values())\n",
+    "\n",
+    "else:\n",
+    "    counts_gps = pickle.load(open(\"results/{}/geo_ctr_gps.pkl\".format(folder_name), \"rb\"))\n",
+    "    region_to_id = counts_gps[\"region_to_id\"]\n",
+    "    id_to_gps = counts_gps[\"id_to_gps\"]\n",
+    "    geo_boundaries = dataset.geo_boundaries\n",
+    "    choro_data = pd.read_csv(dataset.choropleth_filepath)\n",
+    "\n",
+    "    counts = {}\n",
+    "    for region in region_to_id:\n",
+    "        counts[region] = len(region_to_id[region])\n",
+    "    total = sum(counts.values())\n",
+    "    country_count_phrases = []\n",
+    "\n",
+    "\n",
+    "    for country, count in sorted(counts.items(), key=lambda x: x[1], reverse=True):\n",
+    "        country_count_phrases.append(\"{0}: {1}   {2}%\".format(country, count, round(100.*count/total)))\n",
+    "\n",
+    "    colors = [\n",
+    "        'red',\n",
+    "        'blue',\n",
+    "        'gray',\n",
+    "        'darkred',\n",
+    "        'beige',\n",
+    "        'green',\n",
+    "        'purple',\n",
+    "        'pink',\n",
+    "        'green',\n",
+    "        'purple',\n",
+    "        'pink',\n",
+    "    ]\n",
+    "    subregion_counts = {}\n",
+    "    for key in region_to_id:\n",
+    "        filename_arr = region_to_id[key]\n",
+    "        subregion_counts[key] = len(filename_arr)\n",
     "\n",
     "def country_counts_num(topn):\n",
+    "\n",
     "    print(\"Total images: {}\\n\".format(total))\n",
     "    print(\"Country Counts\\n\")\n",
-    "    \n",
+    "\n",
     "    print(\"Top:\\n\")\n",
+    "    topn = min(topn, len(country_count_phrases))\n",
     "    for i in range(topn):\n",
     "        print(country_count_phrases[i])\n",
-    "    \n",
+    "\n",
     "    print(\"\\nBottom:\\n\")\n",
     "    for i in range(topn):\n",
     "        print(country_count_phrases[-1-i])\n",
-    "         \n",
+    "   \n",
     "def subregion_counts_num():\n",
```

Comment:
what does this do for gps coordinates? is it necessary, or should it only be printed if we are in a situation where we are looking at a global dataset?

Reply:
discussed